tests/replication/_base.py

```diff
@@ -154,10 +154,12 @@ def handle_http_replication_attempt(self) -> SynapseRequest:
         self.assertEqual(port, 8765)
 
         # Set up client side protocol
-        client_protocol = client_factory.buildProtocol(None)
+        client_address = IPv4Address("TCP", "127.0.0.1", 1234)
+        client_protocol = client_factory.buildProtocol(client_address)
```

Comment:
Just curious why the change from calling `buildProtocol` with `client_address` as the parameter vs `None`?

Reply:
The address is supposed to be passed, `None` is not a legal value there: https://twistedmatrix.com/documents/current/api/twisted.internet.interfaces.IProtocolFactory.html#buildProtocol

Although now that I check it is supposed to be a tuple version, not an `IAddress` instance. 😢

Reply:
> Was it just not working then, or did it work but you're changing it to be more consistent with the proper usage?

I think (currently) the value never gets used, I was hoping this would fix the bug, but the "real" fix was adding it to the transport. It seemed best to leave it as it was more correct to pass a value here than `None`.

Reply:
Cool thanks for answering!

Reply:
LGTM assuming tests pass

Reply:
> Cool thanks for answering!

No problem!

I think the tl;dr is "It was wrong, but fixing it doesn't fix it. Let's include the fix anyway" 😆